src/extension/serializer.ts

```diff
@@ -1009,16 +1047,395 @@ export class GrpcSerializer extends SerializerBase {
     this.serverReadyListener?.dispose()
     super.dispose()
   }
+}
 
-  public getMaskedCache(cacheId: string): Promise<Uint8Array> | undefined {
-    return this.maskedCache.get(cacheId)
+export class ConnectSerializer extends GrpcSerializerBase {
+  log: ReturnType<typeof getLogger>
+  client: PromiseClient<typeof ParserService>
+  serializerServiceUrl: string
+  protected ready: ReadyPromise
+
+  private serverReadyListener: Disposable | undefined
+
+  constructor(
+    protected context: ExtensionContext,
+    serializerAddress: string,
+    kernel: Kernel,
+  ) {
+    super(context, kernel)
+    this.log = getLogger('ConnectSerializer')
+    this.serializerServiceUrl = serializerAddress
+    this.client = this.createSerializerClient()
+    this.ready = new Promise((resolve) => {
+      resolve()
+    })
   }
 
-  public getPlainCache(cacheId: string): Promise<Uint8Array> | undefined {
-    return this.plainCache.get(cacheId)
+  createSerializerClient = () => {
+    this.log.info(`Serializer: Client pointed to: ${this.serializerServiceUrl}`)
+    // Server options:
+    // (1) pre-existing, started internally by this extension (assuming local execution so it has permissions to start)
+    // (2) pre-existing, started externally, presumably at the this.serializerServiceUrl address
+    //   In both cases we need a key & cert (assuming tls) which may reside in:
+    //   (a) this project: '/path/to/projectRoot/tls'
+    //   (b) the runme binary server default: '~/.config/runme/tls'
+    //   (c) other, manually configured
+
+    // I will assume 2a here
+    const tlsPath = path.normalize(__dirname + '/../tls')
+    const keyPath = `${tlsPath}/key.pem`
+    const certPath = `${tlsPath}/cert.pem`
+
+    let grpcOptions = {
+      baseUrl: this.serializerServiceUrl,
+      httpVersion: '2',
+      nodeOptions: {
+        rejectUnauthorized: false, // allow self-signed certificates
+        cert: fs.readFileSync(certPath),
```

Comment:
What happens if the file doesn't exist?

Reply:
the extension fails to initialize. I will add some error checking here.

Reply:
resolving this. can re-comment if this needs additional attention.